astro/src/content/blog/authorization-airport-analogy.mdx

```diff
@@ -0,0 +1,65 @@
+
+<AuthenticationCodeFlow alt={"Authentication code flow."}/>
+
+## Airport Analogy For Authorization Code Flow
+
+- The user is represented by you (a traveler).
+- The client application is the airport.
+- The Authorization Server is the TSA (Transportation Security Administration). If you are from outside of the United States, you can just think of this as airport security. The airline is also considered part of the authorization server here because they are the ones that issue the boarding pass.
+- The Access Token is the boarding pass.
+- The resource server is the gate your plane leaves from.
+
+<AirportCodeFlow alt={"Airport analogy flow."}/>
+
+So let's walk through this analogy together, from check-in to takeoff.
+
+You (the user) wake up early, caffeinate appropriately, and head to the airport. You've got a destination in mind, let's say maybe beautiful Colorado. You can't just show up and expect to walk straight onto a plane. First, you interact with the airport, our stand-in for the client application. The airport is the one coordinating the trip on your behalf. They know which flight you're taking, what time you need to check in, and where to direct you. They manage the process of getting you to your destination.
+
+When it's time to go through security, the airport directs you to the TSA, which represents the authorization server in our metaphor. This is the critical gatekeeper responsible for verifying your identity. Behind the scenes, the airport has agreed to trust the TSA. You hand over your credentials, usually a government-issued ID and your boarding pass. You can think of the boarding pass as your requested scopes for access to the gates. The TSA doesn't care which airline you flew with last week, they just want to validate that you are you, and that you're permitted to travel.
+
+Once you pass that check, TSA has verified that you are you and you can proceed to the restricted resource. In this case the gates. The analogy breaks down a bit here as the boarding pass now becomes our metaphorical access token and was issued by the airlines and not the TSA. For the purpose of this analogy, We'll lump the TSA and the airline check-in counter into one metaphorical group we'll call 'security & validation.' Together, they handle verifying your identity, issuing your pass, and ensuring you're legit to board. The combination could even be considered multi-factor authentication. In the real authentication flow the authentication server would issue a JWT, which is the actual token used in the authentication flow. This token contains information about who you are, where you're going, and what you're allowed to access. Maybe you're seated in economy or maybe you've got gold status and access to all the snacks and reclining chairs. That's encoded in the token. It's signed, tamper-proof, and TSA doesn't need to follow you around anymore. They trust the pass to do the job.
```

Comment:
I don't think the analogy breaks down. What happens is authentication and then the first level of authorization. 

Check-in: authentication
Security: authorization 1
Boarding: authorization 2